src/main/java/com/basho/riak/client/core/RiakNode.java

```diff
@@ -874,6 +898,25 @@ public void onException(Channel channel, final Throwable t)
         }
     }
 
+    private void checkNetworkAddressCacheSettings()
+    {
+        final String property = Security.getProperty("networkaddress.cache.ttl");
+        if (property == null)
+        {
+            return;
+        }
+
+        int cacheTTL = Integer.parseInt(property);
+
+        if (cacheTTL == -1)
```

Comment:
How about this: if `networkaddress.cache.ttl` is `-1` *and* host names are being used for Riak node addresses, we set the TTL to 60 seconds as recommended everywhere on the internets. A setting could be added to override this behavior.